lib/core/data/post/post_repository_impl.dart

```diff
@@ -66,7 +66,7 @@ class PostRepositoryImpl implements PostRepository {
     if (result.hasException) {
       return Left(Failure());
     }
-    return Right(result.parsedData == true);
+    return Right(PostDto.fromJson(result.parsedData));
```

Comment:
Repository should only return entities not dto. please convert from dto to entites in parseFn of gql